tests/unit/tools/test_unit_mcp.py

```diff
@@ -348,3 +350,91 @@ async def test_agno_mcp_sse() -> None:
+    mock_session.__aexit__.return_value = None
+    mock_session.initialize = AsyncMock()
+    mock_session.list_tools = AsyncMock()
+    # Create a dummy streams context manager
+    dummy_transport = (AsyncMock(), AsyncMock())
+    mock_stdio_context = MagicMock()
+    mock_stdio_context.__aenter__.return_value = dummy_transport
+
+    def mock_stdio_client_func(
+        *args: dict[str, Any], **kwargs: dict[str, Any]
+    ) -> ClientSession:
+        return mock_stdio_context
+
+    # This is fragile because it relies on the 3rd party libs not changing where they import these mcp libs
+    # I am totally open if someone has an idea about a more stable way to do this.
+    where_frameworks_import = [
+        ("agno.tools.mcp.stdio_client", "agno.tools.mcp.ClientSession"),
+        ("agents.mcp.server.stdio_client", "agents.mcp.server.ClientSession"),
+        (
+            "any_agent.tools.mcp.frameworks.langchain.stdio_client",
+            "any_agent.tools.mcp.frameworks.langchain.ClientSession",
+        ),
+        (
+            "llama_index.tools.mcp.client.stdio_client",
+            "llama_index.tools.mcp.client.ClientSession",
+        ),
+        ("mcpadapt.core.stdio_client", "mcpadapt.core.ClientSession"),
+        (
+            "google.adk.tools.mcp_tool.mcp_session_manager.stdio_client",
+            "google.adk.tools.mcp_tool.mcp_session_manager.ClientSession",
+        ),
+    ]
+    for stdio_module, client_module in where_frameworks_import:
+        monkeypatch.setattr(stdio_module, mock_stdio_client_func)
+        monkeypatch.setattr(client_module, lambda *args, **kwargs: mock_session)
+
+    # Use different patches based on the agent framework
+    # Create three mock tools with different names
+    from mcp import Tool as MCPTool
+
+    write_tool = MCPTool(
+        name="write_file", inputSchema={"type": "string", "properties": {}}
+    )
+    read_tool = MCPTool(
+        name="read_file", inputSchema={"type": "string", "properties": {}}
+    )
+    other_tool = MCPTool(
+        name="other_tool", inputSchema={"type": "string", "properties": {}}
+    )
+
+    # Create a ToolList object or mock with the same structure
+    mock_tool_list = MagicMock()
+    mock_tool_list.tools = [
+        write_tool,
+        read_tool,
+        other_tool,
+    ]  # Set the tools attribute directly
+
+    # Mock the list_tools method to return the mock tool list
+    mock_context.list_tools.return_value = mock_tool_list
+    mock_context.initialize = AsyncMock()
+
+
+@pytest.mark.asyncio
+async def test_stdio_tool_filtering(
+    agent_framework: AgentFramework, tmp_path: Path, monkeypatch: Any
+) -> None:
+    setup_mock_server(monkeypatch)
+    mock_stdio_params = MagicMock(spec=MCPStdioParams)
+    mock_stdio_params.command = "test_command"
+    mock_stdio_params.args = ["arg1", "arg2"]
+    mock_stdio_params.tools = ["write_file", "read_file"]
+    mock_stdio_params.client_session_timeout_seconds = 10
```

Comment:
Why do you have to mock this? Can't you just pass an actual MCPStdioParams?

Reply:
Good point, no reason